apps/openmw/mwmechanics/mechanicsmanagerimp.cpp

```diff
@@ -1209,36 +1206,20 @@ namespace MWMechanics
 
         // Did anyone see it?
         bool crimeSeen = false;
-        for (std::vector<MWWorld::Ptr>::iterator it = neighbors.begin(); it != neighbors.end(); ++it)
+        for (const MWWorld::Ptr &neighbor : neighbors)
         {
-            if (*it == player)
-                continue; // skip player
-            if (it->getClass().getCreatureStats(*it).isDead())
+            if (!canReportCrime(neighbor, victim, playerFollowers))
                 continue;
 
-            // Unconsious actor can not report about crime
-            if (it->getClass().getCreatureStats(*it).getKnockedDown())
-                continue;
-
-            if ((*it == victim && victimAware)
-                    || (MWBase::Environment::get().getWorld()->getLOS(player, *it) && awarenessCheck(player, *it) )
+            if ((neighbor == victim && victimAware)
                     // Murder crime can be reported even if no one saw it (hearing is enough, I guess).
                     // TODO: Add mod support for stealth executions!
-                    || (type == OT_Murder && *it != victim))
+                    || (type == OT_Murder && neighbor != victim)
+                    || (MWBase::Environment::get().getWorld()->getLOS(player, neighbor) && awarenessCheck(player, neighbor)))
             {
-                // Crime reporting only applies to NPCs
-                if (!it->getClass().isNpc())
-                    continue;
-
-                if (it->getClass().getCreatureStats(*it).getAiSequence().isInCombat(victim))
-                    continue;
-
-                if (playerFollowers.find(*it) != playerFollowers.end())
-                    continue;
-
```

Comment:
In fact, most of the checks here seem to not have been replaced. There's a lot more red than green here.